app/experimenter/experiments/tests/test_forms.py

```diff
@@ -679,6 +679,16 @@ def test_form_saves_variants(self):
         )
         self.assertEqual(branch2.value, self.data["variants-2-value"])
 
+    def test_form_is_invalid_if_pref_value_do_not_match_pref_type(self):
+        self.data["variants-0-value"] = "42"
+        self.data["variants-1-value"] = "abc"  # no quotes
```

Comment:
@leplatrem This will fail validation for being an invalid JSON blob so I don't think it's properly testing the type case, change this to a valid JSON literal, ie a quoted string and I think this is good to go.